src/gui/EditWidgetIcons.cpp

```diff
@@ -258,9 +289,9 @@ void EditWidgetIcons::startFetchFavicon(const QUrl& url)
 #ifdef WITH_XC_NETWORKING
     m_bytesReceived.clear();
 
+    m_fetchUrl = url;
+
     QNetworkRequest request(url);
-    request.setAttribute(QNetworkRequest::FollowRedirectsAttribute, true);
-    request.setAttribute(QNetworkRequest::RedirectPolicyAttribute, QNetworkRequest::NoLessSafeRedirectPolicy);
```

Comment:
Would it be possible to use these attributes if the Qt version is > 5.9? Likewise in the code above this only perform the redirect checks if the Qt version is older.

#if QT_VERSION >= QT_VERSION_CHECK(5, 9, 0)
// Do attribute stuff
#endif

Reply:
I think it's messier to do the ifdefs. Why not just do it this way until the future when 5.9 is prolific and then move it over to the simpler way?

Reply:
The ifdefs are messier, but it only exposes a small population to the workaround code.

Reply:
I'm not sure I get the argument. From what I read, you're advocating for:

1. having the less readable ifdefs,
2. having multiple possible code paths that are chosen at compile-time,
3. making it harder to debug if something goes wrong (which code path got used at compile time? which Qt version is it?)

Why would you want that instead of a single well-tested and readable code path that all builds use?

Reply:
Well when you put it that way!